File: DailyChallenges/Scala/LeetCode_1704.scala
object Solution {
    def halvesAreAlike(s: String): Boolean = {
        def isVowel(c: Char): Boolean = List('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U') contains c
        s.splitAt(s.length/2) match { case (h1, h2) => h1.count(isVowel) == h2.count(isVowel) }
    }
}

File: DailyChallenges/CPP/LeetCode_1704.cpp
// 1704. Determine if String Halves Are Alike @ 2022/12/01

class Solution {
public:
    bool halvesAreAlike(string s) {
        auto isVol = [](char ch) {
            return set({'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}).count(ch) > 0;};
        
        int h = s.size()/2;
        return count_if(s.begin(), s.begin()+h, isVol) == count_if(s.begin()+h, s.end(), isVol);
    }
};
